### PR TITLE
Fix: Cover CiInfo

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] turns `$_SERVER` into a hard dependency of `CiInfo`, and covers it.

### CODE REVIEW

File: src/TestReporter/Entity/CiInfo.php [hunk 1]
```diff
@@ -4,149 +4,173 @@
 class CiInfo
 {
     /**
+     * @var array
+     */
+    private $info;
+
+    public function __construct(array $server = null)
```
Comment (localheinz) [inline src/TestReporter/Entity/CiInfo.php]: Allowing to inject `$server` here makes it easier to test.

File: src/TestReporter/Entity/CiInfo.php [hunk 2]
```diff
@@ -4,149 +4,173 @@
...
+    {
+        if ($server === null) {
+            $server = $_SERVER;
+        }
+
+        $this->info = $this->infoFrom($server);
```
Comment (localheinz) [inline src/TestReporter/Entity/CiInfo.php]: No need to determine the desired information every time, once should be fine.

File: src/TestReporter/Entity/CiInfo.php [hunk 3]
```diff
@@ -4,149 +4,169 @@
 class CiInfo
 {
     /**
+     * @var array
+     */
+    private $info;
+
+    public function __construct(array $server)
```
Comment (localheinz) [inline src/TestReporter/Entity/CiInfo.php]: Requiring `$server` here allows us to properly test it, and frees this class from needing to access global state.

File: src/TestReporter/Entity/CiInfo.php [hunk 4]
```diff
@@ -4,149 +4,169 @@
...
-        return [ ];
+        return [];
     }
 
     /**
+     * @param array $server
      * @return array
      */
-    protected function travisProperties()
+    protected function travisProperties(array $server)
```
Comment (localheinz) [inline src/TestReporter/Entity/CiInfo.php]: Visibility can probably be reduced to `private`.

File: src/TestReporter/Entity/JsonFile.php [hunk 5]
```diff
@@ -61,7 +61,7 @@ protected function collectGitInfo()
      */
     protected function collectCiServiceInfo()
     {
-        $ciInfo = new CiInfo();
+        $ciInfo = new CiInfo($_SERVER);
```
Comment (localheinz) [inline src/TestReporter/Entity/JsonFile.php]: Injecting `$_SERVER` here now, hehe!

**Comment by dblandin** [comment]
@localheinz This LGTM, but just to clarify, this doesn't sound like a bug to me but rather gets in the way of testing this class properly. Is that your understanding as well?

**Comment by localheinz** [comment]
@dblandin 

Right!

The aim is to get everything solidly covered by tests, and then see about refactoring the existing code!

**Comment by localheinz** [comment]
@dblandin 

Ha, looks like status check for Code Climate is failing, is that a new requirement to make it pass for now?

**Comment by dblandin** [comment]
I approved the Code Climate complexity issues. I assume that we'll clean those up later :)

**Comment by dblandin** [comment]
> is that a new requirement to make it pass for now?

We've made it a required GitHub status check now that collaborators on Code Climate are able to approve failing pull requests.

**Comment by localheinz** [comment]
Thank you, @dblandin!

**Pull request merged**